spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SkipGramFeatureGenerator.java

```diff
@@ -0,0 +1,189 @@
+
+    for (Map.Entry<String, String> entry : config.getFeatureListMapping().entrySet()) {
+      result = loadedModel.setInputCol(entry.getKey() + "-transformed").setOutputCol(entry.getValue())
+        .transform(documentDF);
+      documentDF = result;
+    }
+
+    outputSchema = outputSchema != null ? outputSchema : config.getOutputSchema(inputSchema);
+
+    List<Column> requiredFields = new ArrayList<>();
+    for (Schema.Field field : outputSchema.getFields()) {
+      requiredFields.add(new Column(field.getName()));
+    }
+
+    JavaRDD<Row> transformedRDD =
+      javaSparkContext.parallelize(result.select(JavaConversions.asScalaBuffer(requiredFields).toSeq())
+                                     .collectAsList());
+    JavaRDD<StructuredRecord> output = transformedRDD.map(new Function<Row, StructuredRecord>() {
+      @Override
+      public StructuredRecord call(Row row) throws Exception {
+        StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+        Schema.Field[] strings = outputSchema.getFields().toArray(new Schema.Field[outputSchema.getFields().size()]);
+        for (int i = 0; i < strings.length; i++) {
+          String fieldName = strings[i].getName();
+          Schema schema = strings[i].getSchema();
+          if (config.getFeatureListMapping().values().contains(strings[i].getName())) {
+            double[] doubles = ((Vector) row.get(i)).toArray();
+            builder.set(fieldName, new ArrayList<>(Arrays.asList(ArrayUtils.toObject(doubles))));
+          } else {
+            if (row.get(i) == null && !schema.isNullable()) {
+              throw new IllegalArgumentException(String.format("Null value found for nun-nullable field %s.",
+                                                               fieldName));
+            }
+            Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+            if (type.equals(Schema.Type.ARRAY)) {
+              builder.set(fieldName, row.getList(i));
+            } else if (type.equals(Schema.Type.MAP)) {
+              builder.set(fieldName, row.getJavaMap(i));
+            } else {
+              builder.set(fieldName, row.get(i));
+            }
+          }
+        }
+        return builder.build();
+      }
+    });
+    return output;
+  }
+
+  /**
+   * Config class for SkipGramFeatureGenerator.
+   */
+  public static class FeatureGeneratorConfig extends PluginConfig {
+    @Description("The name of the FileSet to load the model from.")
+    private final String fileSetName;
+
+    @Description("Path of the FileSet to load the model from.")
+    private final String path;
+
+    @Description("List of the input fields to map to the transformed output fields. The key specifies the name of " +
+      "the field to generate feature vector from, with its corresponding value the output column in which the vector " +
+      "would be emitted.")
+    private final String outputColumnMapping;
```

Comment:
there should be a nullable property for the split pattern to use to split input fields. It should default to whitespace.